llama_index/indices/vector_store/retrievers/retriever.py

```diff
@@ -112,6 +112,16 @@ def _retrieve(
                         ] = self._docstore.get_node(  # type: ignore[index]
                             node_id
                         )
+                    # Fix metadata for nodes that have source node in the docstore
+                    elif (
+                        source_node is not None
+                        and source_node.node_id in self._docstore.docs
```

Comment:
`self._docstore.docs` is a pretty expensive operation. Can we call this once and reuse it?

Also, I have no idea how the code gets here when using milvus -- the top level if statement will return false

Reply:
Done! The code applies to both Milvus and Redis vector store since they won't show metadata anyway, so I'm assuming that this fix applies to all the vector store database except the Simple Vector Store.